larpandoracontent/LArHelpers/LArFormattingHelper.cc

```diff
@@ -0,0 +1,184 @@
+
+    void LArFormattingHelper::ResetStyle(std::ostream &stream)
+    {
+        LArFormattingHelper::SetStyle(REGULAR, stream);
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::ResetColour(std::ostream &stream)
+    {
+        LArFormattingHelper::SetColour(DEFAULT, stream);
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::Reset(std::ostream &stream)
+    {
+        LArFormattingHelper::ResetColour(stream);
+        LArFormattingHelper::ResetStyle(stream);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::PrintFormatCharacter(const unsigned int &code, std::ostream &stream)
+    {
+        stream << LArFormattingHelper::GetFormatCharacter(code);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    std::string LArFormattingHelper::GetFormatCharacter(const unsigned int &code)
+    {
+        return ("\x1B[" + std::to_string(code) + "m");
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::PrintHeader(const std::string &title, const unsigned int &width)
+    {
+        LArFormattingHelper::SetStyle(BOLD);
+        std::cout << std::endl << std::string(width, '-') << "\r-" << title << std::endl;
+        LArFormattingHelper::Reset();
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::PrintRule(const unsigned int &width)
+    {
+        LArFormattingHelper::PrintHeader("", width);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    LArFormattingHelper::Table::Table(const std::vector<std::string> &columnTitles, const unsigned int &precision) :
+        m_columnTitles(columnTitles),
+        m_precision(precision)
+    {
+        m_stringstream.precision(m_precision);
+
+        for (const std::string &title : m_columnTitles)
+            m_widths.push_back(title.length());
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    void LArFormattingHelper::Table::CheckForSeparatorColumn()
```

Comment:
Tentative: perhaps unexpected that this will be a non-const function, given the function name? CheckAndSet... instead?

Reply:
Renamed to CheckAndSetSeparatorColumn, and made other member functions const to be explicit